plugin/storage/es/spanstore/writer_test.go

```diff
@@ -48,7 +48,7 @@ func withSpanWriter(fn func(w *spanWriterTest)) {
 		client:    client,
 		logger:    logger,
 		logBuffer: logBuffer,
-		writer:    NewSpanWriter(client, logger, metricsFactory, 0, 0, ""),
+		writer:    NewSpanWriter(SpanWriterParams{Client: client, Logger: logger, MetricsFactory: metricsFactory}),
```

Comment:
One good side about the "params" object for constructors is that it make it act as a named parameter. Bonus points if default values can be omitted, as its the case here.

But I'm still not sure it's the right PR to introduce this pattern, if it's not being used elsewhere already (I don't remember seeing it, but might be wrong).

Reply:
https://github.com/jaegertracing/jaeger/blob/master/cmd/ingester/app/consumer/consumer.go#L54. Search on params